tinygrad/renderer/assembly.py

```diff
@@ -78,6 +85,7 @@ def ptr_ar(root):
 
   matcher = PatternMatcher([
     ({"__name__": "root", "uop": UOps.ALU, "arg": BinaryOps.CMPEQ, "vin": ({"__name__": "x", "dtype": dtypes.bool},{"__name__": "y"})}, eq_rep),
+    ({"__name__": "root", "uop": UOps.ALU, "arg": BinaryOps.ADD, "vin": {}}, mulacc),
```

Comment:
Use a multilevel pattern match here, not a function!

Reply:
No, I mean include the MUL in vin.

Reply:
Oh okay, I see now - that's way better. Updated the PR and also added an option to pass a function to PatternMatcher to eliminate the `if` entirely